WordPress/src/main/java/org/wordpress/android/ui/accounts/PostSignupInterstitialActivity.kt

```diff
@@ -0,0 +1,29 @@
+package org.wordpress.android.ui.accounts
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.post_signup_interstitial_default.*
+import org.wordpress.android.R
+import org.wordpress.android.ui.ActivityLauncher
+
+class PostSignupInterstitialActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.post_signup_interstitial_activity)
+
+        create_new_site_button.setOnClickListener {
+            ActivityLauncher.newBlogForResult(this)
+            finish()
+        }
+
+        add_self_hosted_site_button.setOnClickListener {
+            ActivityLauncher.addSelfHostedSiteForResult(this)
+            finish()
+        }
```

Comment:
The assumption here is that we don't need to listen for the result of any of those calls because `WPMainActivity` would already be on the back stack and would handle the eventual new site automatically by calling [`WPMainActivity.initSelectedSite()`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L1180) inside [`WPMainActivity.onResume()`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L637).

The reason I believe `WPMainActivity` would already be on the back stack is based on the fact that we would be starting `PostSignupInterstitialActivity` from `LoginEpilogueActivity` and `SignupEpilogueActivity`. Those classes are initialized by `LoginActivity` when it calls these methods [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java#L98-L106) and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java#L108-L118), which starts `WPMainActivity` and pass some arguments to it indicating that the epilogue screen should be shown. The other initialization methods on `ActivityLauncher` seem to either be called by `WPMainActivity` itself or are part of other distinct flows around Jetpack or Magic Link, for example. 

If this assumption proves to be incorrect, then I think we could try following the same approach used by `LoginActivity` I described above, where we would listen for the result ourselves and then call `WPMainActivity` with some arguments to instruct it to start the site creation/connection flow as needed.

Another thing to consider is that this topic of whether or not these flows complete correctly might be more related to the next task in the pipeline for this feature ([**Show interstitial screen**](https://github.com/wordpress-mobile/WordPress-Android/issues/10918)). If we realize this would need a more detailed discussion, I'm in favor of letting it be discussed later on the next PR. That way, this one would be focused only on the layouts and the basic behavior of the screen buttons.

Reply:
> If this assumption proves to be incorrect, then I think we could try following the same approach used by LoginActivity I described above, where we would listen for the result ourselves and then call WPMainActivity with some arguments to instruct it to start the site creation/connection flow as needed.

I think you're right on that assumption. If that's not correct, please don't duplicate `initSelectedSite` in there, this one is too sensitive. 

> I'm in favor of letting it be discussed later on the next PR. That way, this one would be focused only on the layouts and the basic behavior of the screen buttons.

+1, we'll see how you integrate this in the signup/login flow and depending on that we can discuss what solution is the best.

Reply:
Thanks for taking the time to review this, @maxme.

> If that's not correct, please don't duplicate `initSelectedSite` in there, this one is too sensitive.

In that case I'd probably try some other, simpler, naive solution first, like just calling `WPMainActivity` the same way I'm doing in [`ActivityLauncher.viewReader()`](https://github.com/renanferrari/WordPress-Android/blob/issue/10923-create-interstitial-screen/WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java#L200-L205) and hoping `onResume()` would do its thing then. Only if _that_ didn't work I'd try to go for some more invasive solution and even then I think we probably wouldn't need to touch `initSelectedSite()` in any way, so I wouldn't worry about that. 😉I appreciate you pointing this out anyway.

> +1, we'll see how you integrate this in the signup/login flow and depending on that we can discuss what solution is the best.

Great! 👍 So I guess there isn't much left to do here. We can probably remove the _Do Not Merge_ tag now and wait for further reviews.